cmd/controller/main.go

```diff
@@ -449,6 +482,59 @@ func validateResource(request resource.Quantity, limit resource.Quantity, resour
 	return validationErrors
 }
 
+func validatePorts(minPort, maxPort int32, portRanges map[string]portallocator.PortRange) []error {
+	validationErrors := make([]error, 0)
+	portErrors := validatePortRange(minPort, maxPort, "")
+	validationErrors = append(validationErrors, portErrors...)
+	for k, r := range portRanges {
+		portErrors = validatePortRange(r.MinPort, r.MaxPort, k)
+		validationErrors = append(validationErrors, portErrors...)
+
+	}
+
+	if len(validationErrors) > 0 {
+		return validationErrors
+	}
+
+	keys := make([]string, 0, len(portRanges))
+	values := make([]portallocator.PortRange, 0, len(portRanges))
+	for k, v := range portRanges {
+		keys = append(keys, k)
+		values = append(values, v)
+	}
+
+	for i, pr := range values {
+		if overlaps(minPort, maxPort, pr.MinPort, pr.MaxPort) {
+			validationErrors = append(validationErrors, errors.Errorf("port range %s overlaps with min/max port", keys[i]))
+		}
+		for j := i + 1; j < len(values); j++ {
+			if overlaps(minPort, maxPort, pr.MinPort, pr.MaxPort) {
+				validationErrors = append(validationErrors, errors.Errorf("port range %s overlaps with min/max port of range %s", keys[i], keys[j]))
+			}
+		}
+	}
+	return validationErrors
+}
+
+func validatePortRange(minPort, maxPort int32, rangeName string) []error {
+	validationErrors := make([]error, 0)
+	var rangeCtx string
+	if rangeName != "" {
+		rangeCtx = " for port range " + rangeName
+	}
+	if minPort <= 0 || maxPort <= 0 {
+		validationErrors = append(validationErrors, errors.New("min Port and Max Port values are required"+rangeCtx))
+	}
+	if maxPort < minPort {
+		validationErrors = append(validationErrors, errors.New("max Port cannot be set less that the Min Port"+rangeCtx))
+	}
+	return validationErrors
+}
+
+func overlaps(minA, maxA, minB, maxB int32) bool {
+	return math.Max(float64(minA), float64(minB)) < math.Min(float64(maxA), float64(maxB))
```

Comment:
nit: Go 1.21 introduced generic https://pkg.go.dev/builtin#max, which avoids the cast and the math import.